infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/utils/InfinispanRemoteTestHelper.java

```diff
@@ -68,6 +69,30 @@ public long getNumberOfAssociations(SessionFactory sessionFactory) {
 		return totalCount;
 	}
 
+	@Override
+	public long getNumberOfAssociations(Session session) {
+		final InfinispanRemoteDatastoreProvider datastoreProvider = getProvider( session.getSessionFactory() );
```

Comment:
shouldn't this method be simply `return getNumberOfAssociation( session.getSessionFactory );`?

Reply:
The main idea here is to handle the fact that size and query APIs are not transactional. This is a big issue for all test that try to count within transactions. The workaround that I used here is use the pesistent context data of a specific session to *enrich* the result with local transaction transient data. The algoritmh is very complex, for this reason I'm not sure to merge it. Unfortunately this one is the only way that I found to count entities and associations. :(